modules/core/src/cuda/gpu_mat.cu

```
@@ -583,4 +594,35 @@ void cv::cuda::GpuMat::convertTo(OutputArray _dst, int rtype, double alpha, doub
     funcs[sdepth][ddepth](reshape(1), dst.reshape(1), alpha, beta, stream);
 }
 
+void cv::cuda::convertFp16Cuda(InputArray _src, OutputArray _dst, Stream& stream)
+{
+    GpuMat src = _src.getGpuMat();
+    int ddepth = 0;
+
+    switch(src.depth())
+    {
+    case CV_32F:
+        ddepth = CV_16S;
+        break;
+    case CV_16S:
+        ddepth = CV_32F;
+        break;
+    default:
+        return;
```

I think it will be better to raise an error here, eg.

```
CV_Error(cv::StsUnsupportedFormat, "Unsupported input depth");
```